### DISCUSSION TITLE
Auto generate operations files based on schema

### DISCUSSION THREAD

**Discussion by luisrudge** [dotansimha/graphql-code-generator#5922] [Q&A] [answered]
```graphql
type Query {
  getOrders(offset: Int = 0, limit: Int = 50): PaginatedOrderList
  getOrderById(id: Int!): Order
}
type Mutation {
  addTracking(
    id: String!
    trackingNumber: String!
  ): AddTracking
}
```

I already have all my possible operations defined. Why do I still need to create another file with the contents of the operations?

```graphql
query getOrders(
  $limit: Int=50
  $offset: Int=0
) {
  getOrders(
    limit: $limit
    offset: $offset
  ) {
    pageInfo {
      total
    }
    results {
      id
      state
      created
    }
  }
}
```

Is there a way to generate all the operations based on my schema?

**Comment by ardatan** [comment, marked as answer]
We have this function builds operations by root field name;
https://github.com/ardatan/graphql-tools/blob/master/packages/utils/src/build-operation-for-field.ts

You can see the use cases here in the unit tests;
https://github.com/ardatan/graphql-tools/blob/master/packages/utils/tests/build-operation-node-for-field.spec.ts

I hope this helps you. If anyone wants to build a plugin based on these, we'd love to accept a PR!

**Comment by luisrudge** [comment]
@ardatan thanks!
